ksql-engine/src/main/java/io/confluent/ksql/function/udf/string/Substring.java

```diff
@@ -21,21 +21,35 @@
 
 @UdfDescription(name = "substring",
     author = "Confluent",
-    description = "returns a substring of the passed in value")
+    description = "Returns a substring of the passed in value")
 public class Substring {
 
-  @Udf(description = "Returns a string that is a substring of this string. The"
-      + " substring begins with the character at the specified startIndex and"
-      + " extends to the end of this string.")
-  public String substring(final String value, final int startIndex) {
-    return value.substring(startIndex);
+  @Udf(description = "Returns a substring of str that starts at pos "
+                     + "and continues to the end of the string")
+  public String substring(
+      final String str,
+      final int pos) {
+    final int start = getStartIndex(str, pos);
+    return str.substring(start);
   }
 
-  @Udf(description = "Returns a string that is a substring of this string. The"
-      + " substring begins with the character at the specified startIndex and"
-      + " extends to the character at endIndex -1.")
-  public String substring(final String value, final int startIndex, final int endIndex) {
-    return value.substring(startIndex, endIndex);
+  @Udf(description = "Returns a substring of str that starts at pos and is of length len")
```

Comment:
was expecting to see some shiny new @UdfParameter decoration here to help teh user know which arg is the length and which one the startPos ? :-)

Reply:
Now on master branch - so annotations added!